### DISCUSSION TITLE
Work status on Harmony OS

### DISCUSSION THREAD

**Discussion by Freed-Wu** [RikkaApps/Shizuku#431] [General]
I haven't found any issue or discussion about the information that run shizuku in Harmony OS so I create this discussion.
I take a try at a harmony OS 3.1.0 device, which looks similar to Android 12. Some weird things are:

1. There is not any wireless debug option in developer mode.
2. Try `adb shell sh /storage/emulated/0/Android/data/moe.shizuku.privileged.api/start.sh` in another machine, if I disconnect them, `shizuku` will stop.
3. `rish` will return

```
Request timeout. The connection between the current app (com.termux) and Shizuku app may be blocked by your system. Please disable all battery optimization features for both current app (com.termux) and Shizuku app.
```

I try to allow `launch setting -> run in background` in `termux` setting, and `shizuku` doesn't have `launch setting` in its settings. However it cannot work.

**Comment by pongfcnkl** [comment]
Using another phone, with this project and an OTG cable, you can activate Shizuku via USB.
https://github.com/nohajc/termux-adb
